suzieq/poller/worker/services/service.py

```diff
@@ -572,9 +595,28 @@ def clean_data_common(self, processed_data, raw_data):
 
     async def commit_data(self, result: Dict, namespace: str, hostname: str):
         """Write the result data out"""
+        # pylint: disable=protected-access
         records = []
         key = f'{namespace}-{hostname}'
-        prev_res = self.previous_results.get(key, [])
+        prev_res = self.previous_results.get(key, None)
+        # No data previously polled with this service from the the
+        # current namespace and hostname. Check if the datastore contains some
+        # information about
+        filter_df = f"hostname.str.match('{hostname}') and " \
+                    f"namespace.str.match('{namespace}')"
+        if prev_res is None:
+            df = self._db_access.read(
+                self.schema_table._table,
```

Comment:
Since the function `_get_db_access` may return None. Shouldn't we add a check here for self._db_access?

Reply:
We use db_access only when we actually use a DB. When we _get_db_access() returns None we are not using a dB, so we'll never enter this function